rpc/src/transports/web_socket.rs

```diff
@@ -194,40 +194,60 @@ impl Transport for WebSocketTransport {
                     return Ok(message);
                 } else {
                     // Ignore messages that are not binary
-                    error!("Received message is not binary");
+                    error!("> WebSocketTransport > Received message is not binary");
                     return Err(TransportError::Internal);
                 }
             }
             Ok(_) => {
-                debug!("Nothing yet")
+                debug!("> WebSocketTransport > Nothing yet")
             }
-            _ => {
-                error!("Failed to receive message");
+            Err(err) => {
+                error!(
+                    "> WebSocketTransport > Failed to receive message {}",
+                    err.to_string()
+                );
             }
         }
-        debug!("Closing transport...");
+        debug!("> WebSocketTransport > Closing transport...");
         self.close().await;
         Ok(TransportEvent::Close)
     }
 
     async fn send(&self, message: Vec<u8>) -> Result<(), TransportError> {
         let message = Message::binary(message);
-        self.write.lock().await.send(message).await?;
-        Ok(())
+        match self.write.lock().await.send(message).await {
+            Err(err) => {
+                error!(
+                    "> WebSocketTransport > Error on sending in a ws connection {}",
+                    err.to_string()
+                );
+
+                match err {
+                    tokio_tungstenite::tungstenite::Error::ConnectionClosed
+                    | tokio_tungstenite::tungstenite::Error::AlreadyClosed => {
+                        return Err(TransportError::Closed);
+                    }
+                    _ => return Err(TransportError::Internal),
```

Comment:
nit: I think this might be a more readable alternative:
```suggestion
                use tokio_tungstenite::tungstenite::Error::*;

                let err = match err {
                    ConnectionClosed | AlreadyClosed => TransportError::Closed,
                    _ => TransportError::Internal,
                };
                Err(err)
```

Reply:
I like it. Thx 💯